### PR TITLE
Refactor skflow

### PR DESCRIPTION
**Pull request by ilblackdragon**
So far:
- Moved ConfigAddon -> RunConfig and capture more about run (tf_master and tf_random_seed).
- Switched skflow trainer to use layers/optimizers and removed Trainer classes
- Switch to gfile for file operations.
- Move creation of placeholders into DataFeeders
- Correcting / enabling examples.

### CODE REVIEW

File: tensorflow/contrib/skflow/python/skflow/estimators/base.py [hunk 1]
```diff
@@ -480,8 +481,9 @@ def restore(cls, path, config_addon=None):
         if not os.path.exists(model_def_filename):
             raise ValueError("Restore folder doesn't contain model definition.")
         # list of parameters that are allowed to be reconfigured
-        reconfigurable_params = ['config_addon']
-        with open(model_def_filename) as fmodel:
+        reconfigurable_params = ['_config']
```
Comment (terrytangyuan) [inline tensorflow/contrib/skflow/python/skflow/estimators/base.py]: We'll probably need to check this in restore/save tests.

**Comment by ilblackdragon** [comment]
@tensorflow-jenkins test this please.

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Comment by caisq** [comment]
@martinwicke @vrv Okay to merge? We need the changes to fix the pylit sanity errors.

**Comment by vrv** [comment]
There were changes so we have to re-test.

@tensorflow-jenkins test this please

**Pull request merged**